sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -273,6 +276,16 @@ export class ManagementClient extends LinkEntity {
   private async _init(): Promise<void> {
     try {
       if (!this._isMgmtRequestResponseLinkOpen()) {
+        // Check that the connection isn't in the process of closing.
```

Comment:
Looks like this block is in 3 spots.

It looks like we really just want the connectionContext to have a single method that lets us `await connectionContext.readyToOpen()` or something of that nature?

Reply:
`readyToOpen`, I like the sounds of that. I'll take a look and see if this still works after some consolidation!